control_plane/attachment_service/src/service.rs

```diff
@@ -2754,6 +3290,29 @@ impl Service {
         &self,
         config_req: NodeConfigureRequest,
     ) -> Result<(), ApiError> {
+        let _node_lock = self.node_locks.exclusive(config_req.node_id).await;
+
+        let activate = {
+            let locked = self.inner.read().unwrap();
+            let Some(node) = locked.nodes.get(&config_req.node_id) else {
+                return Err(ApiError::NotFound(
+                    anyhow::anyhow!("Node {} not registered", config_req.node_id).into(),
+                ));
+            };
+
+            if !node.is_available()
+                && matches!(config_req.availability, Some(NodeAvailability::Active))
+            {
+                Some(node.clone())
+            } else {
+                None
+            }
```

Comment:
This code largely duplicates the stuff below it. Can you call `node_activate_reconcile` on `AvailabilityTransition::ToActive`
insted? Should yield the same result since we do the arc swap at the end.

Reply:
The awkward part here is that we only get the AvailabilityTransition as a result of setting the node's availability, but we don't want to do that until after the reconciliation is done.  But this can probably be cleaned up somehow...

Reply:
So the multi-step-ness of this is indeed necessary, because we're taking the self.inner lock twice, with some async work in between (node_activate_reconcile).

I've cleaned it up a bit to make the stages a bit more obvious, and calculate the AvailabilityTransition up top instead of getting it later when doing set_availability.

Reply:
I see. Looks like the node lock ensures the availability transition computed at the start remains valid across the await. Perhaps worthy of a comment? Up to you though.

Reply:
Yep.  Needed to merge in main anyway, so added a comment while I was at it.